src/js/views/right_panel_view.js

```diff
@@ -702,6 +708,44 @@
             return false;
         },
 
+        handle_user_zoom: function(event){
+            if (event.type === "keyup" && event.which !== 13) {
+                return;     // Ignore keyups except 'Enter'
+            }
+
+            // get the current entered value 
+            var value = event.target.value;
+            if (isNaN(value)) {
+                return;
+            }
+
+            // get min/max and restrict the user input between min/max
+            var minVal = $("#vp_zoom_slider").slider("option", "min");
+            var maxVal = $("#vp_zoom_slider").slider("option", "max");
+
+            if (value < minVal) value = minVal;
+            if (value > maxVal) value = maxVal;
+
+            // update the slider position
+            $("#vp_zoom_slider").slider({'value': value});
+
+            // update the preview image
+            this.update_img_css(value, 0, 0);
+
+            // update the figure image
+            this.zoom_avg = value;
+            var to_save = {'zoom': value};
```

Comment:
@Rdornier Just testing this, and I realise that the `value` never gets converted from a string to an Integer.
JavaScript works OK with the string as it casts it on the fly for e.g. `value < minVal` etc.
But when we save it to Figure JSON model, it gets saved as a string (since we don't have any validation of the model).
This means that when you subsequently select multiple panels, the average zoom values aren't calculated properly. E.g. `"100" + "200" = "100200"` !

Reply:
@will-moore Ok, I've just added a commit to convert the value coming from the UI from string to int (in the listener function and the slider implementation)